Move the assembly property information to the msbuild info article (#22918)
dotnet/docs: docs/core/project-sdk/msbuild-props.md

</PropertyGroup>
```

## Assembly info generation properties

- [GenerateAssemblyInfo](#generateassemblyinfo)

> - [GeneratedAssemblyInfoFile](#generatedassemblyinfofile)

Reviewer: Can you alphabetize this index?

- [GenerateAssemblyCompanyAttribute](#generateassemblycompanyattribute)
- [GenerateAssemblyConfigurationAttribute](#generateassemblyconfigurationattribute)
- [GenerateAssemblyCopyrightAttribute](#generateassemblycopyrightattribute)
- [GenerateAssemblyDescriptionAttribute](#generateassemblydescriptionattribute)
- [GenerateAssemblyFileVersionAttribute](#generateassemblyfileversionattribute)
- [GenerateAssemblyInformationalVersionAttribute](#generateassemblyinformationalversionattribute)
- [GenerateAssemblyProductAttribute](#generateassemblyproductattribute)
- [GenerateAssemblyTitleAttribute](#generateassemblytitleattribute)
- [GenerateAssemblyVersionAttribute](#generateassemblyversionattribute)
- [GenerateNeutralResourcesLanguageAttribute](#generateneutralresourceslanguageattribute)

### GenerateAssemblyInfo

Controls `AssemblyInfo` attribute generation for the project. The default value is `true`. Use `false` to disable generation of the file:

```xml
<PropertyGroup>
  <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
</PropertyGroup>
```

> The name of the file generated is controlled by the [GeneratedAssemblyInfoFile](#generatedassemblyinfofile) setting.

Reviewer: ```suggestion
The [GeneratedAssemblyInfoFile](#generatedassemblyinfofile) setting controls the name of the generated file.
```

> When this value is `true`, project properties are transformed into `AssemblyInfo` attributes. The following table lists the project properties that generate the attributes, and the property that can disable it:

Reviewer: ```suggestion
When the `GenerateAssemblyInfo` value is `true`, project properties are transformed into `AssemblyInfo` attributes. The following table lists the project properties that generate the attributes, and the properties that can disable that generation:
```

| Property               | Attribute                                                      | Property to disable                                                                               |
|------------------------|----------------------------------------------------------------|---------------------------------------------------------------------------------------------------|

> | `Company`              | <xref:System.Reflection.AssemblyCompanyAttribute>              | [`GenerateAssemblyCompanyAttribute`](#generateassemblycompanyattribute)                           |

Reviewer: Should we add a line with a link to this section from https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#package-properties?

| `AssemblyVersion`      | <xref:System.Reflection.AssemblyVersionAttribute>              | [`GenerateAssemblyVersionAttribute`](#generateassemblyversionattribute)                           |
| `NeutralLanguage`      | <xref:System.Resources.NeutralResourcesLanguageAttribute>      | [`GenerateNeutralResourcesLanguageAttribute`](#generateneutralresourceslanguageattribute)         |

Notes about these settings:

- `AssemblyVersion` and `FileVersion` default to the value of `$(Version)` without the suffix. For example, if `$(Version)` is `1.2.3-beta.4`, then the value would be `1.2.3`.
- `InformationalVersion` defaults to the value of `$(Version)`.
- If the `$(SourceRevisionId)` property is present, it's appended to `InformationalVersion`. You can disable this behavior using `IncludeSourceRevisionInInformationalVersion`.
- `Copyright` and `Description` properties are also used for NuGet metadata.
- `Configuration`, which defaults to `Debug`, is shared with all MSBuild targets. You can set it via the `--configuration` option of `dotnet` commands, for example, [dotnet pack](../tools/dotnet-pack.md).

#### Migrating from .NET Framework

> .NET Framework project templates created a code file with these assembly info attributes set, typically located at *.\Properties\AssemblyInfo.cs* or *.\Properties\AssemblyInfo.vb*. Newer SDK-style projects generate this file for you based on the project settings. **You can't have both.** When porting your code to .NET 5 (and .NET Core 3.1) or later, do one of the following:

Reviewer: ```suggestion
.NET Framework project templates create a code file with these assembly info attributes set. The file is typically located at *.\Properties\AssemblyInfo.cs* or *.\Properties\AssemblyInfo.vb*. SDK-style projects generate this file for you based on the project settings. **You can't have both.** When porting your code to .NET 5 (and .NET Core 3.1) or later, do one of the following:
```

> - Disable the generation of temporary code file that contains the assembly info attributes by setting `GenerateAssemblyInfo` to `false`. This enables you to keep your *AssemblyInfo* file.

Reviewer: ```suggestion
- Disable the generation of the temporary code file that contains the assembly info attributes by setting `GenerateAssemblyInfo` to `false`. This enables you to keep your *AssemblyInfo* file.
```

> - Migrate the settings in the `AssemblyInfo` file to the project, and delete the `AssemblyInfo` file.

Reviewer: ```suggestion
- Migrate the settings in the `AssemblyInfo` file to the project file, and delete the `AssemblyInfo` file.
```

### GeneratedAssemblyInfoFile

> The relative or absolute path of the generated assembly info file. Defaults to a file named *[project-name].AssemblyInfo.[cs|vb]* in the `$(IntermediateOutputPath)` (usually the *obj*) directory.

Reviewer: ```suggestion
The property defines the relative or absolute path of the generated assembly info file. Defaults to a file named *[project-name].AssemblyInfo.[cs|vb]* in the `$(IntermediateOutputPath)` (usually the *obj*) directory.
```

```xml
<PropertyGroup>

>   <GeneratedAssemblyInfoFile>false</GeneratedAssemblyInfoFile>

Reviewer: I don't think `false` is correct in this example.
